common/http.cpp

```diff
@@ -0,0 +1,158 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#include <common/common.hpp>
+#include <common/http.hpp>
+
+#include <cstdlib>
+
+namespace mender {
+namespace http {
```

Comment:
Originally my thinking was to not put this in `common` at all (which means the file locations are all wrong), but this is a tricky corner case. Here's the thing: The auth daemon is supposed to have HTTP and HTTPS support. The client however, is supposed to have only HTTP support, because it will connect to the server via the auth daemon through a local network port. But making this split on the code level is difficult without duplicating code, so I think I'll backtrack on my plan and make it "common" as you say.